external/llvm-project/mlir/lib/Dialect/Tosa/Transforms/TosaPartition.cpp

```diff
@@ -444,59 +484,60 @@ class TosaPartitionPass : public TosaPartitionBase<TosaPartitionPass> {
+      if (func->hasAttr(attributeName()))
         continue;
 
       int count = 0;
       // (Problems with node mismatches and unexpected uses if we have the
       // candidates list at module level.)
       std::vector<OutliningCandidate> candidates;
-      auto callback = [&](tosa::Conv2DOp convOp) {
+      auto callback = [&](Operation *op) {
+        if (!isAnchorOp(op))
+          return WalkResult::advance();
+        Operation *anchorOp = op;
         auto strCount =
             std::string("_outlined_part_") + std::to_string(count++);
 
-        // Given a Conv2DOp, gather all the element-wise ops that are reachable
-        // from its results, contiguously.
+        // Given a Conv2DOp (or other anchor op), gather all the
+        // element-wise ops that are reachable from its results,
+        // contiguously.
         //
-        // The ops after the Conv2D are "second" ops.
-        // inputNodes gathers what will become the parameters of the outlined
-        // function;  initially it's the Conv2D's arguments, and it accumulates
-        // arguments to other ops that don't come from inside the outlined
-        // function.
-        // resultNodes will become the results of the outlined function.  It
-        // starts with Conv2D's result(s) and gains the results of each new
-        // secondOp.  When all a resultNode's users can be determined to lie
-        // within the outlined function, it's removed from the set.
+        // The ops after the anchor are "trailing" ops.
         //
-        // These are SetVectors because we test with contains() a lot, but still
-        // want to preserve order.
-        SetVector<Operation *> secondOps;
+        // inputNodes gathers what will become the parameters of the
+        // outlined function;  initially it's the anchor's arguments,
+        // and it accumulates arguments to other ops that don't come
+        // from inside the outlined function.
+        //
+        // resultNodes will become the results of the outlined function.
+        // It starts with the anchor's result(s) and gains the results
+        // of each new trailingOp.  When all a resultNode's users can be
+        // determined to lie within the outlined function, it's removed
+        // from the set.
+        //
+        // These are SetVectors because we test with contains() a lot,
+        // but still want to preserve order.
+        SetVector<Operation *> trailingOps;
         SetVector<Value> inputNodes;
-        SetVector<Value> resultNodes(convOp->getResults().begin(),
-                                     convOp->getResults().end());
+        SetVector<Value> resultNodes(anchorOp->getResults().begin(),
+                                     anchorOp->getResults().end());
 
         // Grab a useful set of leading ops, like we do for trailing.
-        // Let's limit it to only first arguments, with single uses.
-        SmallVector<Operation *> frontOps;
-        if (!nofront) {
```

Comment:
The "nofront" test is absorbed into isLeadingOp() -- see the with-options version below.